contracts/service-registry/src/contract/migrations/mod.rs

```diff
@@ -1,16 +1,187 @@
-use axelar_wasm_std::migrate_from_version;
+#[cfg(test)]
+mod legacy_state;
+
+use axelar_wasm_std::nonempty::Usize;
+use axelar_wasm_std::{migrate_from_version, IntoContractError};
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
 use cosmwasm_std::{DepsMut, Empty, Env, Response};
+use error_stack::ResultExt;
+use service_registry_api::error::ContractError;
+
+use crate::state::{services, update_authorized_verifier_count};
 
 pub type MigrateMsg = Empty;
 
+#[derive(thiserror::Error, Debug, PartialEq, IntoContractError)]
+enum MigrationError {
+    #[error("invalid limit")]
+    InvalidLimit,
+}
+
 #[cfg_attr(not(feature = "library"), entry_point)]
-#[migrate_from_version("1.1")]
+#[migrate_from_version("1.2")]
 pub fn migrate(
     deps: DepsMut,
     _env: Env,
     _msg: MigrateMsg,
 ) -> Result<Response, axelar_wasm_std::error::ContractError> {
+    let services = services(
+        deps.storage,
+        None,
+        Usize::try_from(usize::MAX).change_context(MigrationError::InvalidLimit)?,
+    )
+    .change_context(ContractError::ServiceNotFound)?;
+
+    for s in services {
+        let _ = update_authorized_verifier_count(deps.storage, &s.name)?;
+    }
```

Comment:
you expose two new public functions from state that are otherwise not necessary just to feed the output of one into the other. I suggest making it a single call to the state module, and also add a comment on that function that it was only used for migrations and should be removed once it's not needed anymore